phpBB/includes/acp/acp_extensions.php

```diff
@@ -65,22 +77,22 @@ function main()
 		* @var	string	ext_name		Extension name from request
 		* @var	int		safe_time_limit	Safe limit of execution time
 		* @var	int		start_time		Start time
+		* @var	string	tpl_name		Template file to load; leave empty to continue execution, filled in if ready to finish
 		* @since 3.1.11-RC1
 		*/
 		$u_action = $this->u_action;
-		$vars = array('action', 'u_action', 'ext_name', 'safe_time_limit', 'start_time');
+		$tpl_name = '';
+		$vars = array('action', 'u_action', 'ext_name', 'safe_time_limit', 'start_time', 'tpl_name');
 		extract($this->phpbb_dispatcher->trigger_event('core.acp_extensions_run_action', compact($vars)));
 
-		// Cancel action
-		if ($request->is_set_post('cancel'))
-		{
-			$action = 'list';
-			$ext_name = '';
-		}
+		// In case they have been updated by the event
+		$this->u_action = $u_action;
+		$this->tpl_name = $tpl_name;
 
-		if (in_array($action, array('enable', 'disable', 'delete_data')) && !check_link_hash($request->variable('hash', ''), $action . '.' . $ext_name))
+		// If tpl_name was set by the prior event, we are done
```

Comment:
Somewhat unexpected side effect. Not sure I like this. Couldn't you simply modify the action to have pretty much the same happening?